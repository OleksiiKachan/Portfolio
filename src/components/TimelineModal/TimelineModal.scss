@import '../../assets/style/variables';

.modal {
  position: fixed;
  display: inline-block;
  z-index: 2;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);
}

.modal-main {
  position: fixed;
  background-color: $color-black;
  color: $color-white;
  min-width: 80%;
  height: auto;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  box-shadow: 2px 2px 20px 0px rgba(0, 0, 0, 0.29);
}

.modal-closeButton {
  float: right;
  margin: 10px;
}

@media screen and (max-width: 825px) {
  .modal-main {
    top: 50%;
    left: 50%;
    transform: translate(-50% -50%);
  }
}
