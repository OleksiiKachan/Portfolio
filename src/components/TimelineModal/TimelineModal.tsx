import React, { Component, createRef } from 'react';
import ReactDOM from 'react-dom';
import classNames from 'classnames';
import { EducationType, ExperienceType } from '../../lib/types';
import EducationPage from './EducationPage';
import ExperiencePage from './ExperiencePage';
import './TimelineModal.scss';

const icon_close = require('../../assets/icons/icon_close.svg');

/*
    <TimelineModal
      handleClose=() => void
      item={}
      type=''
      modalRootID=''
    />
*/

type PropsType = {
  handleClose: () => void;
  item?: EducationType | ExperienceType;
  type: string;
  modalRootID: string;
};

export default class TimelineModal extends Component<PropsType> {
  private modalRef = createRef<HTMLDivElement>();
  state = { show: false };

  componentDidMount() {
    document.addEventListener('mousedown', e => this.handleClickOutside(e));
  }

  componentWillUnmount() {
    document.removeEventListener('mousedown', e => this.handleClickOutside(e));
  }

  handleClickOutside(event: MouseEvent) {
    if (
      this.modalRef &&
      this.modalRef.current &&
      !this.modalRef.current.contains(event.target as Node)
    ) {
      this.props.handleClose();
    }
  }
  render() {
    const portalContent = [
      <div className={classNames('modal')}>
        <section className="modal-main" ref={this.modalRef}>
          <img
            src={icon_close}
            alt="close icon"
            onClick={this.props.handleClose}
            className={classNames('modal-closeButton')}
          />
          {this.props.type === 'education' ? (
            <EducationPage item={this.props.item as EducationType} />
          ) : (
            <ExperiencePage item={this.props.item as ExperienceType} />
          )}
        </section>
      </div>,
    ];

    const portalRoot = document.getElementById(this.props.modalRootID);
    const portal = portalRoot
      ? [ReactDOM.createPortal(portalContent, portalRoot)]
      : portalContent;

    return [...portal];
  }
}
