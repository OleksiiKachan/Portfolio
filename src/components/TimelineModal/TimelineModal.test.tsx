import React from 'react';
import { shallow } from 'enzyme';
import TimelineModal from './';

describe('TimelineModal', () => {
  it('Snapshot', () => {
    const component = shallow(
      <TimelineModal
        handleClose={() => {}}
        show={true}
        item={{
          id: 'kpi',
          school:
            'National Technical University of Ukraine "Kyiv Polytechnic Institute"',
          degree: 'Bachelor of Software Engineering',
          department:
            'Faculty of Informatics and Computer Science, Department of Technical Cybernetics',
          program: 'Software Engineering',
          dates: '2013-2017',
        }}
        type="education"
        modalRootID="root"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
