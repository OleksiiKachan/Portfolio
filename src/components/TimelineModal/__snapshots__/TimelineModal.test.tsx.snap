// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineModal Snapshot 1`] = `
<div
  className="modal display-block"
>
  <section
    className="modal-main"
  >
    <img
      className="modal-closeButton"
      onClick={[Function]}
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "icon_close.svg",
        }
      }
    />
    <_default
      item={
        Object {
          "dates": "2013-2017",
          "degree": "Bachelor of Software Engineering",
          "department": "Faculty of Informatics and Computer Science, Department of Technical Cybernetics",
          "id": "kpi",
          "program": "Software Engineering",
          "school": "National Technical University of Ukraine \\"Kyiv Polytechnic Institute\\"",
        }
      }
    />
  </section>
</div>
`;
